### PR TITLE
sign the image urls

### PR DESCRIPTION
**Pull request by AWare**
## What does this change?
Signs the ImageBlockElements urls.
## Screenshots
# 🖼🖊
### Does this affect other platforms?

- [X] dotcom rendering

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Comment by philmcmahon** [comment]
Hey @AWare see what you think if my latest commit - it looks like this:

screen shot 2018-11-19 at 18 59 08

File: common/app/layout/WidthsByBreakpoint.scala [hunk 1]
```diff
@@ -211,6 +211,14 @@ object ContentWidths {
     def thumbnail: WidthsByBreakpoint = unused
     def immersive: WidthsByBreakpoint = unused
     def halfwidth: WidthsByBreakpoint = unused
+
+    def all: Map[String, WidthsByBreakpoint] = Map(
```
Comment (AWare) [inline common/app/layout/WidthsByBreakpoint.scala]: We've got `Role`s now that can do the strings job here :D

File: common/app/model/dotcomrendering/pageElements/PageElement.scala [hunk 2]
```diff
@@ -71,7 +74,9 @@ object Sponsorship {
   }
 }
 
+//noinspection ScalaStyle
```
Comment (AWare) [inline common/app/model/dotcomrendering/pageElements/PageElement.scala]: I just boosted the max complexity here so we might want to change it back if we want to just exclude this.

**Comment by AWare** [comment]
Github won't let me approve this officially. But 👍

File: common/app/views/support/ImageProfile.scala [hunk 3]
```diff
@@ -269,28 +284,26 @@ object ImgSrc extends Logging with implicits.Strings {
...
   }
 
   def srcsetForProfile(
     profile: ImageProfile,
     imageContainer: ImageMedia,
     hidpi: Boolean
-  ): String =
-    s"${profile.bestSrcFor(imageContainer).get} ${profile.width.get * (if (hidpi) 2 else 1)}w"
+  ): SrcSet =
+    SrcSet(profile.bestSrcFor(imageContainer).get, profile.width.get * (if (hidpi) 2 else 1))
```
Comment (kelvin-chappell) [inline common/app/views/support/ImageProfile.scala]: `get` is prone to explode if there's a possibility of undefined values.  Would be safer to map over `profile.bestSrcFor(imageContainer)` and `profile.width` and give some default value where one of them isn't defined.

File: common/app/views/support/ImageProfile.scala [hunk 4]
```diff
@@ -269,28 +284,26 @@ object ImgSrc extends Logging with implicits.Strings {
...
 
-  def srcsetForProfile(profile: ImageProfile, path: String, hidpi: Boolean): String =
-    s"${ImgSrc(path, profile)} ${profile.width.get * (if (hidpi) 2 else 1)}w"
+  def srcsetForProfile(profile: ImageProfile, path: String, hidpi: Boolean): SrcSet =
+    SrcSet(ImgSrc(path, profile), profile.width.get * (if (hidpi) 2 else 1))
```
Comment (kelvin-chappell) [inline common/app/views/support/ImageProfile.scala]: Same as above here.

File: common/app/model/dotcomrendering/pageElements/PageElement.scala [hunk 5]
```diff
@@ -105,12 +110,24 @@ object PageElement {
...
-        imageDataFor(element),
-        element.imageTypeData.flatMap(_.displayCredit),
-        Role(element.imageTypeData.flatMap(_.role))
-      ))
+      case Image =>
+        val signedAssets = element.assets.zipWithIndex
+          .map { case (a, i) => ImageAsset.make(a, i) }
+        val imageSources: Seq[ImageSource] = BodyMedia.all.map {
+          case (weighting, widths) =>
+            val srcSet = widths.breakpoints.map { b =>
```
[the remark this answers is not in the document]
Comment (philmcmahon) [reply common/app/model/dotcomrendering/pageElements/PageElement.scala]: thanks1

File: common/app/views/support/ImageProfile.scala (hunk 3 above)
Comment (philmcmahon) [reply common/app/views/support/ImageProfile.scala]: yep good shout!

**Comment by philmcmahon** [comment]
@AWare we should get this out on Monday, I've given it a quick check on code and there don't seem to be any obvious issues

**Review by kelvin-chappell — APPROVED** [review]
👍

**Pull request merged**